Importamos las librerías Pandas y Numpy

In [65]:
import pandas as pd
import numpy as np

### Step 2. Tomaremos el fichero en [esta URL](https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv) y lo leeremos como un dataframe.

Pista: [read_csv](https://numpy.org/doc/stable/user/absolute_beginners.html#importing-and-exporting-a-csv)

In [66]:
url = "https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv"

# Leer el archivo CSV desde la URL y convertirlo en un DataFrame
df_chipotle = pd.read_csv(url, sep="\t")

# Mostrar las primeras filas del DataFrame
df_chipotle.head(25)


order_id  quantity                              item_name  \
0          1         1           Chips and Fresh Tomato Salsa   
1          1         1                                   Izze   
2          1         1                       Nantucket Nectar   
3          1         1  Chips and Tomatillo-Green Chili Salsa   
4          2         2                           Chicken Bowl   
5          3         1                           Chicken Bowl   
6          3         1                          Side of Chips   
7          4         1                          Steak Burrito   
8          4         1                       Steak Soft Tacos   
9          5         1                          Steak Burrito   
10         5         1                    Chips and Guacamole   
11         6         1                   Chicken Crispy Tacos   
12         6         1                     Chicken Soft Tacos   
13         7         1                           Chicken Bowl   
14         7         1                    Chips and Guacamole   
15         8         1  Chips and Tomatillo-Green Chili Salsa   
16         8         1                        Chicken Burrito   
17         9         1                        Chicken Burrito   
18         9         2                            Canned Soda   
19        10         1                           Chicken Bowl   
20        10         1                    Chips and Guacamole   
21        11         1                       Barbacoa Burrito   
22        11         1                       Nantucket Nectar   
23        12         1                        Chicken Burrito   
24        12         1                                   Izze   

                                   choice_description item_price  
0                                                 NaN     $2.39   
1                                        [Clementine]     $3.39   
2                                             [Apple]     $3.39   
3                                                 NaN     $2.39   
4   [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5   [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                                 NaN     $1.69   
7   [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8   [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
9   [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25   
10                                                NaN     $4.45   
11  [Roasted Chili Corn Salsa, [Fajita Vegetables,...     $8.75   
12  [Roasted Chili Corn Salsa, [Rice, Black Beans,...     $8.75   
13  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...    $11.25   
14                                                NaN     $4.45   
15                                                NaN     $2.39   
16  [Tomatillo-Green Chili Salsa (Medium), [Pinto ...     $8.49   
17  [Fresh Tomato Salsa (Mild), [Black Beans, Rice...     $8.49   
18                                           [Sprite]     $2.18   
19  [Tomatillo Red Chili Salsa, [Fajita Vegetables...     $8.75   
20                                                NaN     $4.45   
21  [[Fresh Tomato Salsa (Mild), Tomatillo-Green C...     $8.99   
22                               [Pomegranate Cherry]     $3.39   
23  [[Tomatillo-Green Chili Salsa (Medium), Tomati...    $10.98   
24                                       [Grapefruit]     $3.39

In [67]:
#1ero voy a limpiar la columna item_price quitandole el signo $
df_chipotle_clean = df_chipotle.copy() 

df_chipotle_clean['item_price'] = df_chipotle_clean['item_price'].str.replace('$', '').astype(float)

df_chipotle_clean.head(25)


order_id  quantity                              item_name  \
0          1         1           Chips and Fresh Tomato Salsa   
1          1         1                                   Izze   
2          1         1                       Nantucket Nectar   
3          1         1  Chips and Tomatillo-Green Chili Salsa   
4          2         2                           Chicken Bowl   
5          3         1                           Chicken Bowl   
6          3         1                          Side of Chips   
7          4         1                          Steak Burrito   
8          4         1                       Steak Soft Tacos   
9          5         1                          Steak Burrito   
10         5         1                    Chips and Guacamole   
11         6         1                   Chicken Crispy Tacos   
12         6         1                     Chicken Soft Tacos   
13         7         1                           Chicken Bowl   
14         7         1                    Chips and Guacamole   
15         8         1  Chips and Tomatillo-Green Chili Salsa   
16         8         1                        Chicken Burrito   
17         9         1                        Chicken Burrito   
18         9         2                            Canned Soda   
19        10         1                           Chicken Bowl   
20        10         1                    Chips and Guacamole   
21        11         1                       Barbacoa Burrito   
22        11         1                       Nantucket Nectar   
23        12         1                        Chicken Burrito   
24        12         1                                   Izze   

                                   choice_description  item_price  
0                                                 NaN        2.39  
1                                        [Clementine]        3.39  
2                                             [Apple]        3.39  
3                                                 NaN        2.39  
4   [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98  
5   [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...       10.98  
6                                                 NaN        1.69  
7   [Tomatillo Red Chili Salsa, [Fajita Vegetables...       11.75  
8   [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...        9.25  
9   [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...        9.25  
10                                                NaN        4.45  
11  [Roasted Chili Corn Salsa, [Fajita Vegetables,...        8.75  
12  [Roasted Chili Corn Salsa, [Rice, Black Beans,...        8.75  
13  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...       11.25  
14                                                NaN        4.45  
15                                                NaN        2.39  
16  [Tomatillo-Green Chili Salsa (Medium), [Pinto ...        8.49  
17  [Fresh Tomato Salsa (Mild), [Black Beans, Rice...        8.49  
18                                           [Sprite]        2.18  
19  [Tomatillo Red Chili Salsa, [Fajita Vegetables...        8.75  
20                                                NaN        4.45  
21  [[Fresh Tomato Salsa (Mild), Tomatillo-Green C...        8.99  
22                               [Pomegranate Cherry]        3.39  
23  [[Tomatillo-Green Chili Salsa (Medium), Tomati...       10.98  
24                                       [Grapefruit]        3.39

In [68]:
df_chipotle_clean.describe()

order_id     quantity   item_price
count  4622.000000  4622.000000  4622.000000
mean    927.254868     1.075725     7.464336
std     528.890796     0.410186     4.245557
min       1.000000     1.000000     1.090000
25%     477.250000     1.000000     3.390000
50%     926.000000     1.000000     8.750000
75%    1393.000000     1.000000     9.250000
max    1834.000000    15.000000    44.250000

### Step 3. Veamos los tipos de datos. ¿Podríamos obtener el producto de mayor precio?

In [69]:
# Calcula el precio unitario
df_chipotle_clean['unit_price'] = df_chipotle_clean['item_price'] / df_chipotle_clean['quantity']

# Encuentra el item_name con el precio unitario más alto
max_unit_price_row = df_chipotle_clean.loc[df_chipotle_clean['unit_price'].idxmax()]

# Extrae el item_name
max_unit_name_item = max_unit_price_row['item_name']
max_unit_price_item = max_unit_price_row['item_price']

print(f"El item_name con el item_price más alto por unidad es: {max_unit_name_item} {max_unit_price_item}")


El item_name con el item_price más alto por unidad es: Steak Salad Bowl 23.78


In [70]:
df_chipotle_clean

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description  item_price  \
0                                                   NaN        2.39   
1                                          [Clementine]        3.39   
2                                               [Apple]        3.39   
3                                                   NaN        2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98   
...                                                 ...         ...   
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...       11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...       11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...       11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...        8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...        8.75   

      unit_price  
0           2.39  
1           3.39  
2           3.39  
3           2.39  
4           8.49  
...          ...  
4617       11.75  
4618       11.75  
4619       11.25  
4620        8.75  
4621        8.75  

[4622 rows x 6 columns]

### Step 4. ¿Qué productos cuestan más de $10?

In [73]:
### Step 4. ¿Qué productos cuestan más de $10?

# Filtra los registros con unit_price mayor a 10
df_filtered = df_chipotle_clean[df_chipotle_clean['unit_price'] > 10]

# Agrupa por 'item_name' y 'unit_price' y cuenta las ocurrencias
grouped_df = df_filtered.groupby(['item_name', 'unit_price']).count()

grouped_df


order_id  quantity  choice_description  \
item_name             unit_price                                           
Barbacoa Bowl         11.48              6         6                   6   
                      11.49              1         1                   1   
                      11.75             20        20                  20   
Barbacoa Burrito      11.08              1         1                   1   
                      11.48              1         1                   1   
                      11.75             22        22                  22   
Barbacoa Crispy Tacos 11.48              1         1                   1   
                      11.75              3         3                   3   
Barbacoa Salad Bowl   11.89              5         5                   5   
Barbacoa Soft Tacos   11.48              1         1                   1   
                      11.75              7         7                   7   
Carnitas Bowl         11.08              3         3                   3   
                      11.48              7         7                   7   
                      11.75             22        22                  22   
Carnitas Burrito      11.08              1         1                   1   
                      11.48              2         2                   2   
                      11.75             16        16                  16   
Carnitas Crispy Tacos 11.75              2         2                   2   
Carnitas Salad Bowl   11.89              4         4                   4   
Carnitas Soft Tacos   11.75              3         3                   3   
Chicken Bowl          10.58              1         1                   1   
                      10.98              1         1                   1   
                      10.98             49        49                  49   
                      11.25            231       231                 231   
Chicken Burrito       10.58              4         4                   4   
                      10.98             45        45                  45   
                      11.25            131       131                 131   
Chicken Crispy Tacos  10.98              1         1                   1   
                      11.25             14        14                  14   
Chicken Salad         10.98              2         2                   2   
Chicken Salad Bowl    11.25             51        51                  51   
Chicken Soft Tacos    10.98              4         4                   4   
                      11.25             22        22                  22   
Steak Bowl            11.08              2         2                   2   
                      11.48             10        10                  10   
                      11.75             78        78                  78   
Steak Burrito         11.08              5         5                   5   
                      11.48             21        21                  21   
                      11.75            101       101                 101   
Steak Crispy Tacos    11.75             11        11                  11   
Steak Salad Bowl      11.89             19        19                  19   
Steak Soft Tacos      11.48              3         3                   3   
                      11.75              6         6                   6   
Veggie Bowl           10.98              4         4                   4   
                      11.25             41        41                  41   
Veggie Burrito        10.98              4         4                   4   
                      11.25             33        33                  33   
Veggie Salad Bowl     11.25             10        10                  10   
Veggie Soft Tacos     11.25              2         2                   2   

                                  item_price  
item_name             unit_price              
Barbacoa Bowl         11.48                6  
                

### Step 4.1: ¿Y cuántos pedidos se han hecho con un producto de más de 10$? ¿Es lo mismo?

In [81]:
# Filtra los registros con unit_price mayor a 10
df_filtered = df_chipotle_clean[df_chipotle_clean['unit_price'] > 10]

# Agrupa por 'item_name' y 'unit_price' y cuenta las ocurrencias
grouped_df = df_filtered.groupby(['item_name', 'unit_price']).count()

grouped_df


order_id  quantity  choice_description  \
item_name             unit_price                                           
Barbacoa Bowl         11.48              6         6                   6   
                      11.49              1         1                   1   
                      11.75             20        20                  20   
Barbacoa Burrito      11.08              1         1                   1   
                      11.48              1         1                   1   
                      11.75             22        22                  22   
Barbacoa Crispy Tacos 11.48              1         1                   1   
                      11.75              3         3                   3   
Barbacoa Salad Bowl   11.89              5         5                   5   
Barbacoa Soft Tacos   11.48              1         1                   1   
                      11.75              7         7                   7   
Carnitas Bowl         11.08              3         3                   3   
                      11.48              7         7                   7   
                      11.75             22        22                  22   
Carnitas Burrito      11.08              1         1                   1   
                      11.48              2         2                   2   
                      11.75             16        16                  16   
Carnitas Crispy Tacos 11.75              2         2                   2   
Carnitas Salad Bowl   11.89              4         4                   4   
Carnitas Soft Tacos   11.75              3         3                   3   
Chicken Bowl          10.58              1         1                   1   
                      10.98              1         1                   1   
                      10.98             49        49                  49   
                      11.25            231       231                 231   
Chicken Burrito       10.58              4         4                   4   
                      10.98             45        45                  45   
                      11.25            131       131                 131   
Chicken Crispy Tacos  10.98              1         1                   1   
                      11.25             14        14                  14   
Chicken Salad         10.98              2         2                   2   
Chicken Salad Bowl    11.25             51        51                  51   
Chicken Soft Tacos    10.98              4         4                   4   
                      11.25             22        22                  22   
Steak Bowl            11.08              2         2                   2   
                      11.48             10        10                  10   
                      11.75             78        78                  78   
Steak Burrito         11.08              5         5                   5   
                      11.48             21        21                  21   
                      11.75            101       101                 101   
Steak Crispy Tacos    11.75             11        11                  11   
Steak Salad Bowl      11.89             19        19                  19   
Steak Soft Tacos      11.48              3         3                   3   
                      11.75              6         6                   6   
Veggie Bowl           10.98              4         4                   4   
                      11.25             41        41                  41   
Veggie Burrito        10.98              4         4                   4   
                      11.25             33        33                  33   
Veggie Salad Bowl     11.25             10        10                  10   
Veggie Soft Tacos     11.25              2         2                   2   

                                  item_price  
item_name             unit_price              
Barbacoa Bowl         11.48                6  
                

### Step 4.2: ¿Y cuántos pedidos se han hecho de más de 10$? ¿Es lo mismo?

### No es lo mismo, ya que aqui estamos hablando de solo del pedido de más de 10$

### Step 4.3: ¿Y en cuántos pedidos se ha pagado más de 10$ por un mismo producto? ¿Es lo mismo?

No es lo mismo, ya que la busqueda se centrara  por item_name dentro del dataframe del punto 4, productos_filtrados_cantidad_precio

### Step 5. ¿Qué precio tiene cada producto en distintos pedidos? ¿Hay productos con varios precios?

### Step 6. Ordena el dataframe en base al nombre de producto (item name)

### Step 7. ¿Cuantas veces se ha pedido los productos más caros?

### Step 8. Veamos para el caso de Veggie Salad Bowl. Extrae esa información.